### ISSUE TITLE
OCRmyPDF crashes in convert_to_pdfa when OCRing certain PDF files

### ISSUE THREAD

**Issue by Apolyxon** [jbarlow83/OCRmyPDF#497]
OCRmyPDF crashes in the last step of converting a PDF (convert_to_pdfa). If I split this PDF with some external tool, this does not happen and the command runs without any problems. 

The PDF is generated by a book scanner in a library (Bookeye). All scans from these scanners have this problem. 

**Call Stack**
```
ocrmypdf._sync -   ERROR - An exception occurred while executing the pipeline Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/ocrmypdf/9.6.0/libexec/lib/python3.7/site-packages/ocrmypdf/_sync.py", line 391, in run_pipeline
    exec_concurrent(context)
  File "/home/linuxbrew/.linuxbrew/Cellar/ocrmypdf/9.6.0/libexec/lib/python3.7/site-packages/ocrmypdf/_sync.py", line 320, in exec_concurrent
    pdf = post_process(pdf, context)
  File "/home/linuxbrew/.linuxbrew/Cellar/ocrmypdf/9.6.0/libexec/lib/python3.7/site-packages/ocrmypdf/_sync.py", line 175, in post_process
    pdf_out = convert_to_pdfa(pdf_out, ps_stub_out, context)
  File "/home/linuxbrew/.linuxbrew/Cellar/ocrmypdf/9.6.0/libexec/lib/python3.7/site-packages/ocrmypdf/_pipeline.py", line 697, in convert_to_pdfa
    if pdf_file.docinfo:

TypeError: length not defined for object
```

```bash
 ocrmypdf -v 2 'test1.pdf' 'test2.pdf'
```

**Example file**
[test1.pdf](https://github.com/jbarlow83/OCRmyPDF/files/4255478/test1.pdf)

Please check any or all that apply about the test file:

- [X] This is the input file
- [X] The file contains no personal or confidential information
- [ ] I am the copyright holder for this file
- [ ] I permit this file to be included in the OCRmyPDF test suite under the CC-BY-SA 4.0 license
- [ ] I am not the copyright holder, but this file is available under a free software license

**Expected behavior**
OCRmyPDF should not crash. The PDF looks normal, viewable and not corrupted.

**System:**

- OS: Windows with Linux Subsystem
- OCRmyPDF Version: 9.6.0

**Comment by jbarlow83** [comment]
Another day, another oddly formed PDF.

In test1.pdf ``pdf.trailer.Info``, which should normally contain a DocumentInfo dictionary, is instead a duplicate reference to the XML data normally stored in ``pdf.root.Metadata`` which is also present. It is therefore non-conforming.

The file also claims to be PDF/A-2A compliant, but fails to validate with veraPDF, because the XMP metadata block contains unrecognized schemas. The complaint follows:

```xml
<rule specification="ISO 19005-2:2011" clause="6.6.2.3" testNumber="7" status="failed" passedChecks="0" failedChecks="3">
            <description>All properties specified in XMP form shall use either the predefined schemas defined in the XMP Specification,
                        ISO 19005-1 or this part of ISO 19005, or any extension schemas that comply with 6.6.2.3.2.</description>
            <object>XMPProperty</object>
            <test>(isPredefinedInXMP2005 == true || isDefinedInMainPackage == true || isDefinedInCurrentPackage == true) &amp;&amp; isValueTypeCorrect == true</test>
            <check status="failed">
              <context>root/document[0]/metadata[0](182 0 obj PDMetadata)/XMPPackage[0]/Properties[15](http://www.scan2net.com/xmp/1.0/ - s2n:model)</context>
            </check>
            <check status="failed">
              <context>root/document[0]/metadata[0](182 0 obj PDMetadata)/XMPPackage[0]/Properties[13](http://www.scan2net.com/xmp/1.0/ - s2n:FirmwareVersion)</context>
            </check>
            <check status="failed">
              <context>root/document[0]/metadata[0](182 0 obj PDMetadata)/XMPPackage[0]/Properties[14](http://www.scan2net.com/xmp/1.0/ - s2n:serial)</context>
            </check>
          </rule>
```

There is a second failure which may be the metadata issue or something else.

```xml

<rule specification="ISO 19005-2:2011" clause="6.7.3" testNumber="1" status="failed" passedChecks="0" failedChecks="1">
            <description>The logical structure of the conforming file shall be described by a structure hierarchy rooted in the StructTreeRoot entry 
                        of the document's Catalog dictionary, as described in ISO 32000-1:2008, 14.7.</description>
            <object>PDDocument</object>
            <test>StructTreeRoot_size == 1</test>
            <check status="failed">
              <context>root/document[0]</context>
            </check>
          </rule>
```

In any event, you can manually fix the file by removing the incorrect /Info entry:

```python
import pikepdf
p = pikepdf.open('test1.pdf')
del p.trailer.Info
p.save('test1-fixed.pdf')
```

The metadata is still preserved, since the problem an incorrect duplicate reference to the metadata. A fix for this rare case was added since it seems that these files come from a website that probably produces many files with this problem.

**Comment by Apolyxon** [comment]
Thank you!

These files are generated by a book scanner. These scanners are used in university libraries all over Germany for scanning books.

**Comment by jbarlow83** [comment]
Fixed in v9.6.1
